src/libmain/progress-bar.cc

```diff
@@ -338,6 +350,7 @@ class ProgressBar : public Logger
     {
         auto nextWakeup = std::chrono::milliseconds::max();
 
+        if (state.paused) return nextWakeup;
```

Comment:
should this go after `haveUpdate`?

Reply:
I don't think so. After resuming, the bar should display all the updates which came during the pause.

Reply:
Yeah, but having haveUpdate set will cause the updateThread to attempt the draw call every 50ms. I think it would be better to clear haveUpdate here and then call `update()` in `resume()`? This will also cause a force redraw after the resume.

Reply:
Sure.